File: CS/GrowthWare.Framework/Common/CryptoUtilityException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace GrowthWare.Framework.Common
{
    /// <summary>
    /// Created to distinguish errors created in the CryptoUtility class.
    /// </summary>
    /// <remarks></remarks>
    [Serializable()]
    public class CryptoUtilityException : Exception
    {
        		[System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1032:ImplementStandardExceptionConstructors")]
		public CryptoUtilityException(){}

		/// <summary>
		/// Calls base method
		/// </summary>
		/// <param name="message">string</param>
		public CryptoUtilityException(string message):base(message)
		{
			
		}

		/// <summary>
		/// Calls base method
		/// </summary>
		/// <param name="message">string</param>
		/// <param name="innerException">Exception</param>
		public CryptoUtilityException(string message, Exception innerException):base(message, innerException)
		{
		
		}

		/// <summary>
		/// Calls base method
		/// </summary>
		/// <param name="info"></param>
		/// <param name="context"></param>
        protected CryptoUtilityException(SerializationInfo info, StreamingContext context)
            : base(info, context)
		{
			// Implement type-specific serialization constructor logic.
		}
    }
}

File: Net_Framework/VB/GrowthWare.Framework/Common/CryptoUtilityException.vb
Imports System.Runtime.Serialization

Namespace Common
    ''' <summary>
    ''' Created to distinguish errors created in the data access layer.
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable()>
    Public Class CryptoUtilityException
        Inherits Exception

        Public Sub New()

        End Sub

        ''' <summary>
        ''' Calls base method
        ''' </summary>
        ''' <param name="message">String</param>
        ''' <remarks></remarks>
        Public Sub New(ByVal message As String)
            MyBase.New(message)
        End Sub

        ''' <summary>
        ''' Calls base method
        ''' </summary>
        ''' <param name="message">String</param>
        ''' <param name="innerException">Exception</param>
        ''' <remarks></remarks>
        Public Sub New(ByVal message As String, ByVal innerException As Exception)
            MyBase.New(message, innerException)
        End Sub

        ''' <summary>
        ''' Calls base method
        ''' </summary>
        ''' <param name="info"></param>
        ''' <param name="context"></param>
        ''' <remarks></remarks>
        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
            MyBase.New(info, context)
        End Sub
    End Class

End Namespace
